lib/class-conversionprocessor.php

```diff
@@ -95,10 +98,16 @@ public function get_all_unconverted_ids() {
 				-- Select desired post types.
 				WHERE post_type IN ( {$types_placeholders} )
 				AND post_status IN ( {$statuses_placeholders} )
+				AND ID BETWEEN %d AND %d
 				-- Filter out post which are already in blocks.
 				AND post_content NOT LIKE '<!-- wp:%'
 				ORDER BY ID DESC ;",
-				array_merge( $post_types, $post_statuses )
+				[
+					...$post_types,
```

Comment:
I am probably missing something, but I think we are defining only two placeholder values for `$wpdb->prepare` but passing more.

`WHERE post_type IN ( {$types_placeholders} )` is directly passing the values so no placeholder there. The same goes for `AND post_status IN ( {$statuses_placeholders} )`.

I think we need to skip the `...$post_types` and `...$post_statuses` leaving only:

```
[
    $min_post_id_to_process,
    $max_post_id_to_process,
]
```

Seems to me that before this change, the placeholder values were not being used either 🙂

Reply:
It is definitely confusing to read, but it is correct. WP does not make SQL easy like that!

The `$types_placeholders` and `$statuses_placeholders` will hold something like: `%s` or `%s,%s` (depending on how many status/types there are. The `$post_types` and `$post_statuses` are then spread into the args array for the prepare statement. Right now there will never be more than one of each, but #131 will add the possibility to have multiple.

Maybe best explained with a debugger screenshot:
![CleanShot 2024-05-07 at 12 57 07@2x](https://github.com/Automattic/newspack-content-converter/assets/193283/23755345-1d86-4155-a839-821457b61250)

Reply:
You are absolutely correct!

I misread the definitions of `$types_placeholders` and `$statuses_placeholders`. Thought they hold the actual values instead of placeholders.. 🤦‍♂️

Sorry for having you spend time on this 🙇 

Everything looks good to me now 🙏